src/nn.h

```diff
@@ -352,6 +364,7 @@ NN_EXPORT int nn_setsockopt (int s, int level, int option, const void *optval,
     size_t optvallen);
 NN_EXPORT int nn_getsockopt (int s, int level, int option, void *optval,
     size_t *optvallen);
+NN_EXPORT int nn_setopt (int option, const void *optval, size_t optvallen);
```

Comment:
This does not feel like a common use-case, which would make me lean toward the compile-time option. The cost of introducing this as a run-time option is increasing the surface area of the libnanomsg API, without a substantial ROI on the benefit.

`nn_setopt` might be revisited if there existed more run-time options that added even more value to the existence of `nn_setopt`, but for now doesn't seem like the best option.

It is of course intentionally limiting to set this at compile time -- but trading flexibility for usability *seems* more desirable here.